spinalcordtoolbox/deepseg/core.py

```diff
@@ -22,71 +22,12 @@
 DEFAULTS = {
     'thr': 0.9,
-    'largest': 1,
-    'fill_holes': 1,
+    'largest': 0,
+    'fill_holes': 0,
+    'remove_small': '0vox'
 }
 
 
-def postprocess(nii_seg, options):
-    """
-    Wrapper to apply postprocessing on the segmentation, depending on user's, metadata or default options.
-
-    :param nii_seg: nibabel: Segmentation
-    :param options: dict: Parameters for postprocessing, including keys such as: threshold, keep_largest_object.
-    :return: postprocessed segmentation
-    """
-
-    def threshold(nii_seg, thr):
-        """Threshold the prediction. For no threshold, set 'thr' to 0."""
-        logger.info("Threshold: {}".format(thr))
-        if thr:
-            nii_seg = imed.postprocessing.threshold_predictions(nii_seg, thr)
-        return nii_seg
-
-    def keep_largest_objects(nii_seg, n_objects):
-        """Only keep the n largest objects."""
-        logger.info("Keep largest objects: {}".format(n_objects))
-        if n_objects > 1:
-            # TODO: implement the thing below.
-            NotImplementedError("For now, the algorithm can only remove the largest object, no more than that.")
-        # Make sure input is binary. If not, skip with verbose.
-        if np.array_equal(nii_seg.get_fdata(), nii_seg.get_fdata().astype(bool)):
-            # Fetch axis corresponding to superior-inferior direction
-            # TODO: move that code in image
-            affine = nii_seg.get_header().get_best_affine()
-            code = nib.orientations.aff2axcodes(affine)
-            if 'I' in code:
-                axis_infsup = code.index('I')
-            elif 'S' in code:
-                axis_infsup = code.index('S')
-            else:
-                raise ValueError(
-                    "Neither I nor S is present in code: {}, for affine matrix: {}".format(code, affine))
-            nii_seg = imed.postprocessing.keep_largest_object_per_slice(nii_seg, axis=axis_infsup)
-        else:
-            logger.warning("Algorithm 'keep largest object' can only be run on binary segmentation. Skipping.")
-        return nii_seg
-
-    def fill_holes(nii_seg):
-        """Fill holes"""
-        logger.info("Fill holes")
-        # Make sure input is binary. If not, skip with verbose.
-        if np.array_equal(nii_seg.get_fdata(), nii_seg.get_fdata().astype(bool)):
-            nii_seg = imed.postprocessing.fill_holes(nii_seg)
-        else:
-            logger.warning("Algorithm 'fill holes' can only be run on binary segmentation. Skipping.")
-        return nii_seg
-
-    logger.info("\nProcessing segmentation\n" + "-" * 23)
-    if options['thr']:
-        nii_seg = threshold(nii_seg, options['thr'])
-    if options['largest']:
-        nii_seg = keep_largest_objects(nii_seg, options['largest'])
-    if options['fill_holes']:
-        nii_seg = fill_holes(nii_seg)
-    return nii_seg
-
-
 def segment_nifti(fname_image, folder_model, fname_prior=None, param=None):
```

Comment:
given that there is almost nothing left in this `core.py` file, and that the only thing `segment_nifti()` does is call ivadomed, maybe we can just get rid of `core.py` and move the ivadomed call to the CLI? @Drulex @joshuacwnewton

Reply:
I very much agree that there's not much reason in having this function at all, especially since the file I/O should be handled at the CLI script level anyway. 

We would need to replace `test_segment_nifti()` in `unit_testing/test_deepseg.py` as well, I believe.

Reply:
@andreanne-lemay: if you want we can take care of this part (since we are more familiar with SCT's codebase). @joshuacwnewton do you mind helping with this?

Reply:
Certainly! I would love to. :slightly_smiling_face: